src/Domain/CoordinateMaps/TimeDependent/RotationMatrixHelpers.cpp

```diff
@@ -80,6 +83,56 @@ Matrix rotation_matrix_deriv(
   return rotation_matrix_deriv;
 }
 
+DataVector quaternion_from_angles(double theta_x, double theta_y,
+                                  double theta_z) {
+  const double cx = cos(theta_x);
+  const double cy = cos(theta_y);
+  const double cz = cos(theta_z);
+  const double sx = sin(theta_x);
+  const double sy = sin(theta_y);
+  const double sz = sin(theta_z);
+  // Rotation matrix calculated from yaw, pitch, and roll angles which are
+  // theta_z, theta_y, and theta_x, respectively.
+  Matrix m{{cz * cy, cz * cy * sx - sz * cx, cz * sy * cx + sz * sx},
+           {sz * cy, sz * sy * sx + cz * cx, sz * sy * cx - cz * sx},
+           {-sy, cy * sx, cy * cx}};
+
+  // If we compare this version of the rotation matrix from angles with the one
+  // above which is a rotation matrix calculated from a quaternion, we can solve
+  // for the components of the quaternion like so:
+  //
+  // 1 + m[0][0] - m[1][1] - m[2][2] = 4q[1]^2
+  //               m[1][0] + m[0][1] = 4q[1]*q[2]
+  //               m[2][0] + m[0][2] = 4q[1]*q[3]
+  //               m[1][2] - m[2][1] = 4q[1]*q[0]
+  //
+  // Now we notice that all these equations have a 4q[1] on the RHS. If, for
+  // some given input angles, the first equation gives us q[1] = 0, then we
+  // cannot use these equations to solve for the scalar part of the quaternion
+  // because we'd have to divide by 0. Thus, we need a different set of
+  // equations in order to find the scalar part. These are what the different
+  // branches below are for; accounting for one component of the quaternion
+  // being zero.
+  const double qx = 0.5 * sqrt(1.0 + m(0, 0) - m(1, 1) - m(2, 2));
+  const double qy = 0.5 * sqrt(1.0 - m(0, 0) + m(1, 1) - m(2, 2));
+  const double qz = 0.5 * sqrt(1.0 - m(0, 0) - m(1, 1) + m(2, 2));
```

Comment:
You've got issues with branch cuts here (the correct values can be negative), and switching equations based on roundoff differences usually indicates bad numerical behavior.

Wikipedia [suggests an algorithm](https://en.wikipedia.org/wiki/Rotation_matrix#Quaternion).  I haven't checked it, but it looks sane.

Add tests that would have caught this.